src/wire/ipv4.rs

```diff
@@ -120,6 +133,53 @@ impl Cidr {
         self.prefix_len
     }
 
+    /// Return the network mask of this IPv4 CIDR.
+    pub fn netmask(&self) -> Address {
+        if self.prefix_len == 0 {
+            return Address([0, 0, 0, 0]);
+        }
+
+        let number = 0xffffffffu32 << (32 - self.prefix_len);
+        let data = [
+            ((number >> 24) & 0xff) as u8,
+            ((number >> 16) & 0xff) as u8,
+            ((number >>  8) & 0xff) as u8,
+            ((number >>  0) & 0xff) as u8,
+        ];
+        Address(data)
+    }
+
+    /// Return the broadcast address of this IPv4 CIDR.
+    pub fn broadcast(&self) -> Address {
+        let network = self.network();
+
+        if network.prefix_len == 32 {
```

Comment:
Actually, this is not right. A /31 subnet or /32 "subnet" [does not have a broadcast address](http://packetlife.net/blog/2008/jun/18/using-31-bit-subnets-on-point-point-links/). So, this function should return an `Option<Address>`.

Reply:
👍